robotiq_force_torque_sensor/nodes/rq_sensor.cpp

```diff
@@ -171,206 +288,41 @@ static robotiq_force_torque_sensor::ft_sensor get_data(void)
-    watchdog_pub = nh.advertise<std_msgs::Empty>("/robotiq_ft300_force_torque_sensor/watchdog", 1);
-    service = nh.advertiseService("robotiq_force_torque_sensor_acc", receiverCallback);
-
-    std_msgs::Bool connection_msg;
-    robotiq_force_torque_sensor::ft_sensor msgStream;
-    geometry_msgs::WrenchStamped wrenchMsg;
-    std_msgs::Empty watchdog_msg;
-    ros::param::param<std::string>("~frame_id", wrenchMsg.header.frame_id, "robotiq_force_torque_frame_id");
-    INT_32 publish_count = 0;
-
-
-    // Publish the connection state
-    connection_msg.data = true;
-    connection_pub.publish(connection_msg);
-    ros::spinOnce();
-
-    ROS_INFO("Starting Sensor");
-
-    msgStream.Fx = 0.0;
-    msgStream.Fy = 0.0;
-    msgStream.Fz = 0.0;
-    msgStream.Mx = 0.0;
-    msgStream.My = 0.0;
-    msgStream.Mz = 0.0;
-    wrenchMsg.wrench.force.x = msgStream.Fx;
-    wrenchMsg.wrench.force.y = msgStream.Fy;
-    wrenchMsg.wrench.force.z = msgStream.Fz;
-    wrenchMsg.wrench.torque.x = msgStream.Mx;
-    wrenchMsg.wrench.torque.y = msgStream.My;
-    wrenchMsg.wrench.torque.z = msgStream.Mz;
-
-    while(ros::ok()) {
-        sensor_pub.publish(msgStream);
-        wrenchMsg.header.stamp = ros::Time::now();
-        wrench_pub.publish(wrenchMsg);
-
-        // Publish the connection state every 1000 iterations
-        if (publish_count % 10 == 0) {
-            watchdog_pub.publish(watchdog_msg);
-        }
-
-        if (publish_count == 1000) {
-            connection_msg.data = true;
-            connection_pub.publish(connection_msg);
-            publish_count = 0;
-        }
-        publish_count++;
-        // Keep publishing rate to 100Hz
-        ros::Duration(0.01).sleep();
-        ros::spinOnce();
-    }
-
-    ROS_INFO("Sensor Stopped");
-
-    // Publish the connection state
-    connection_msg.data = false;
-    connection_pub.publish(connection_msg);
-    ros::spinOnce();
-
-    // unregister publishers
-    sensor_pub.shutdown();
-    wrench_pub.shutdown();
-    watchdog_pub.shutdown();
-    service.shutdown();
-    connection_pub.shutdown();
-}
-
-
-void launchRealSensor(ros::NodeHandle& nh)
+int main(int argc, char **argv)
 {
-    std_msgs::Bool connection_msg;
-    ros::param::param<int>("~max_retries", max_retries_, 100);
-
-    INT_8 bufStream[512];
-    INT_32 publish_count = 0;
-    robotiq_force_torque_sensor::ft_sensor msgStream;
-    bool connected = false;
-    uint8_t created_publishers = 0;
-
-    //If we can't initialize, we return an error
-    connected = try_connection();
+    ros::init(argc, argv, "robotiq_force_torque_sensor");
 
-    //Reads basic info on the sensor
-    if (connected) {
-        connected = try_connection();
-    }
+    std::unique_ptr<SensorBase> sensor_ptr;
 
-    //Starts the stream
-    if (connected) {
-        connected = try_connection();
+    bool simulated;
+    ros::param::param<bool>("/simulate_robot", simulated, false);
```

Comment:
There is a `ros::param::param<T>` that takes 2 args -- path, default value and returns the result. I find that more readable.
```suggestion
    bool simulated = ros::param::param<bool>("/simulate_robot", false);
```